.github/workflows/publish.yml

```diff
@@ -8,10 +8,15 @@ on:
   push:
   pull_request:
 
+permissions:
+  contents: read
+
 jobs:
 
   test_wheel_building:
     # This ensures that a couple of targets work fine in pull requests and pushes
+    permissions:
```

Comment:
@WilliamJamieson is going to test this one in a future downstream release.

Reply:
I think I got all the permission stuff right in astropy/asdf-astropy#134